venom_core/services/model_introspection_activation_path_service.py

```
@@ -185,11 +196,152 @@ def _build_mlp_unavailable_payload(
     }
 
 
+def _compute_cosine_similarity(
+    left_slice: list[float],
+    right_slice: list[float],
+    left_norm: float,
+    right_norm: float,
+) -> float | None:
+    if not left_slice or not right_slice or left_norm <= 0.0 or right_norm <= 0.0:
+        return None
+    dot_product = sum(left * right for left, right in zip(left_slice, right_slice))
```

Ruff (0.15.14)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)